gate.lic

```diff
@@ -0,0 +1,194 @@
+    check_buffs
+    moon = get_moon
+    unless moon
+      send_message(person,"No moons, no gates!")
+      exit
+    end
+    focus_moonbeam(moon)
+    send_message(person,"Locating and shifting...")
+    locate_and_shift(moon, person)
+    unless verify_moon?(moon)
+      send_message(person, "The moon is no longer visible. Try again!")
+      echo "#{moon} is no longer visible. Try again!"
+      exit
+    end
+
+    unless teleport
+      send_message(person,"Here comes the gate...")
+      cast_moongate(moon)
+      send_message(person,"Moongate should be open!")
+
+      if release
+        bput('mana',"#{person} came through a")
+        bput('release moongate','You release your mental hold on the lunar energy that sustains your moongate.')
+      end
+     else
+      cast_teleport(moon)
+    end
+  end
+def cast_moongate(moon)
+    data = {'abbrev' => 'MG'}
+    check_discern(data)
+    cast_spell('moongate', data['mana'], moon)
+  end
+
+  def cast_teleport(moon)
+    data = {'abbrev' => 'teleport'}
+    check_discern(data)
+    cast_spell('teleport', data['mana'], moon)
+  end
+
+  def focus_moonbeam(moon)
+    cast_spell('fm', 1, moon, true)
+  end
+
+  def shift_moonbeam
+    return if check_if_buff_active('Shift Moonbeam', 6)
+    cast_spell('sm', 25, 'self')
+  end
+
+  def locate(person, mana = 25)
+    cast_spell('locate', mana, person)
+  end
+
+  def locate_and_shift(moon, person)
+    shift_moonbeam
+    cast_cv
+    locate(person)
+    send_message(person, "found you!")
+    bput("gesture #{moon} #{person}", 'Roundtime')
+  end
+
+  def cast_cv
+    return if check_if_buff_active('Clear Vision', 4)
+    cast_spell('cv', 1, 'self', true)
+  end
+
+  def verify_moon?(moon, uptime = 4)
+
+    is_vis = UserVars.moons['visible'].include? moon
+    is_time = UserVars.moons[moon]['timer'] >= uptime
+    return is_vis && is_time
+
+  end
+
+  def check_if_buff_active(spell, time)
+    return DRSpells.active_spells[spell] && DRSpells.active_spells[spell] > time
+  end
+def get_moon
+    unless moon = UserVars.moons.find { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 8 }.first
+      unless Script.running? 'moonwatch'
+        echo 'moonwatch is not running. Starting it now'
+        custom_require.call('moonwatch')
+        echo "Run `;e autostart('moonwatch')` to avoid this in the future"
+      end
+
+      unless moon = UserVars.moons.find { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 8 }.first
+        echo "Couldn't find any moons to cast with."
+        UserVars.moons.values[0..2].each { |x| echo x['pretty'] }
+        return false
+      end
+    end
+    return moon
+  end
+
+  def check_buffs
+    wait_for_script_to_complete('buff',['gate'])
+  end
+
+  def send_message(character, message)
+    @lnet.unique_buffer.push("chat to #{character} #{message}")
+  end
+
+  # copied from crossing-trianing
+  def check_discern(data)
```

Comment:
This would be another good thing to commonize

Reply:
agreed. However, these use yaml data which as I understand it the common-* scripts should not include functionality that depends on yaml.